Microsoft.Toolkit.Uwp.UI.Controls/MasterDetailsView/MasterDetailsView.cs

```diff
@@ -96,25 +96,48 @@ protected override void OnApplyTemplate()
         }
 
         /// <summary>
-        /// Fired when the SelectedItem changes.
+        /// Fired when the SelectedIndex changes.
         /// </summary>
         /// <param name="d">The sender</param>
         /// <param name="e">The event args</param>
         /// <remarks>
         /// Sets up animations for the DetailsPresenter for animating in/out.
         /// </remarks>
-        private static void OnSelectedItemChanged(DependencyObject d, DependencyPropertyChangedEventArgs e)
+        private static void OnSelectedIndexChanged(DependencyObject d, DependencyPropertyChangedEventArgs e)
         {
             var view = (MasterDetailsView)d;
 
-            view.OnSelectionChanged(new SelectionChangedEventArgs(new List<object> { e.OldValue }, new List<object> { e.NewValue }));
+            var newValue = (int)e.NewValue == -1 ? null : view.Items[(int)e.NewValue];
```

Comment:
> == -1 [](start = 42, length = 6)

You can replace this check by `< 0` to be safer

Reply:
Fixed :-)